### PR TITLE
feat: Add 'MemoryStore'

### PR DESCRIPTION
**Pull request by Uanid**
Features:
1.  Add `MemoryStore`

Reference Issue: #89

### CODE REVIEW

**Review by Wwwsylvia — CHANGES_REQUESTED** [review]

File: inmemory_store.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+import (
+	"context"
+
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// InMemoryStore is a store that keeps credentials in memory.
+type InMemoryStore struct {
+	store map[string]auth.Credential
```
[the remark this answers is not in the document]
Comment (Uanid) [reply inmemory_store.go]: Oh, it needs to be Go-routine safe. I missed that fact. I've pushed a new commit. I would appreciate it if you could review this.

File: inmemory_store.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	"context"
+	"sync"
+
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// InMemoryStore is a store that keeps credentials in memory.
+type InMemoryStore struct {
+	rwLock sync.RWMutex
+	store  map[string]auth.Credential
```
[the remark this answers is not in the document]
Comment (Uanid) [reply inmemory_store.go:28]: Sorry for the late reply. I've been busy recently.

I'm not familiar with the sync.Map struct, so I used rwLock. I will modify it to sync.Map based on your feedback.

File: inmemory_store.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+)
+
+// InMemoryStore is a store that keeps credentials in memory.
+type InMemoryStore struct {
+	store sync.Map
+}
+
+// NewInMemoryStore creates a new in-memory credentials store.
+func NewInMemoryStore() *InMemoryStore {
+	return &InMemoryStore{store: sync.Map{}}
```
Comment (Wwwsylvia) [inline inmemory_store.go:32]: nit: 
> The zero Map is empty and ready for use.
```suggestion
	return &InMemoryStore{}
```

**Review by Wwwsylvia — APPROVED** [review]
LGTM with nit

**Comment by Wwwsylvia** [comment]
@shizhMSFT Could you also take a look?

File: inmemory_store.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+import (
+	"context"
+	"sync"
+
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// InMemoryStore is a store that keeps credentials in memory.
+type InMemoryStore struct {
```
[the remark this answers is not in the document]
Comment (Uanid) [reply inmemory_store.go]: That might be a good name. Changed it to MemoryStore.

File: memory_store.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	store sync.Map
+}
+
+// NewMemoryStore creates a new in-memory credentials store.
+func NewMemoryStore() *MemoryStore {
+	return &MemoryStore{}
+}
+
+// Get retrieves credentials from the store for the given server address.
+func (is *MemoryStore) Get(_ context.Context, serverAddress string) (auth.Credential, error) {
```
Comment (Wwwsylvia) [inline memory_store.go:36]: nit:  `ms *MemoryStore`?  😀

Comment (Wwwsylvia) [reply memory_store.go:36]: You may want to change the PR description as well.

**Review by Wwwsylvia — APPROVED** [review]
LGTM

File: memory_store.go (hunk 5 above)
Comment (Uanid) [reply memory_store.go]: Thanks for your feedback!

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Comment by Wwwsylvia** [comment]
Merged. Thank you @Uanid again for contributing this!

**Pull request merged**